### PR TITLE
feat(connector): add vonage connector

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add Vonage SMS connector.

## Testing

截屏2024-11-05 10 57 17

截屏2024-11-05 10 57 24

截屏2024-11-05 10 58 59

## Checklist

- [x] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by darcyYe — APPROVED** [review]
LGTM

File: .changeset/moody-numbers-brush.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+"@logto/connector-vonage-sms": major
+---
+
+add vonage sms connector
+
+You can now send SMS messages using the Vonage SMS connector.
+
+To work with this connector, you will need to [sign up for an account](https://developer.vonage.com/en/account/guides/dashboard-management#create-and-configure-a-vonage-account) in Vonage. This will give you an API key and secret that you can use to access the APIs through this connector.
+
+Once you have an account, you can find your API Key and API Secret at the top of the Vonage API Dashboard.
```
Comment (charIeszhao) [inline .changeset/moody-numbers-brush.md:11]: nit:
```suggestion
Once you have an account, you can find your API key and API secret at the top of the Vonage API Dashboard.
```

Comment (charIeszhao) [reply .changeset/moody-numbers-brush.md:11]: If you would like to keep `API Key` then you probably need to change the letter case at line 9

File: packages/connectors/connector-vonage-sms/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Vonage SMS connector
+
+The official Logto connector for Vonage SMS.
+
+## Get started
+
+Vonage is a global communications provider, offering robust cloud-based communication services, including SMS (short message service). The Vonage SMS Connector is a plugin provided by the Logto team to enable Logto end-users to register and sign in to their Logto account via SMS verification codes.
+
+## Set Up in Vonage
```
Comment (charIeszhao) [inline packages/connectors/connector-vonage-sms/README.md:9]:
```suggestion
## Set up in Vonage
```

File: packages/connectors/connector-vonage-sms/logo.svg
Comment (charIeszhao) [inline packages/connectors/connector-vonage-sms/logo.svg:1]: Get a dark logo as well?

Comment (wangsijie) [reply packages/connectors/connector-vonage-sms/logo.svg:1]: Maybe later, it is not easy to find one

Comment (charIeszhao) [reply packages/connectors/connector-vonage-sms/logo.svg:1]: Contact designer. 5 mins thing

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**